src/test/java/seedu/address/testutil/TypicalModules.java

```diff
@@ -42,6 +51,13 @@ private TypicalModules() {
     }
 
     public static List<Module> getTypicalModules() {
+        return new ArrayList<>(Arrays.asList(DISCRETE_MATH,
```

Comment:
I think it might be better to explicitly create the desired list of modules in your test class. This applies to `getModulesWithoutNonGradeAffectingModules()` too. Maybe we can have a generic convenience function that takes modules as varargs and returns arrayList. 

In general no one should use `getTypicalModules` for testing anyway since it is liable to be changed, unless you are absolutely sure all you need are valid modules. In which case it seems like we should rename it `getValidModules` because 'typical' is too vague - there is no 'typical' module.